samcli/lib/list/mapper.py

```diff
@@ -0,0 +1,13 @@
+"""
+Interface for Mappers
+"""
+import abc
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
```

Comment:
Is there a way to also define a Generic as the output type for the mapper. Maybe something like 

```
InputType = TypeVar("InputType")
OutputType = TypeVar("OutputType")

class Mapper(Generic[InputType, OutputType]):
    @abstractmethod
    def map(self, input: InputType) -> OutputType:
        pass
```
I'm not sure about ^, it needs to be confirmed.